x/autopilot/module_ibc.go

```diff
@@ -169,9 +169,31 @@ func (im IBCModule) OnRecvPacket(
 
 	//// At this point, we are officially dealing with an autopilot packet
 
-	// Modify the packet data by replacing the JSON metadata field with a receiver address
-	// to allow the packet to continue down the stack
-	tokenPacketData.Receiver = autopilotMetadata.Receiver
+	// Update the reciever in the packet data so that we can pass the packet down the stack
+	// (since the "receiver" may have technically been a full JSON memo)
+	switch routingInfo := autopilotMetadata.RoutingInfo.(type) {
```

Comment:
Wdyt of simplifying like so?

```
// Set default receiver
tokenPacketData.Receiver = autopilotMetadata.Receiver

// Override receiver for autopilot liquid stake with non-empty IBC receiver
if routingInfo, ok := autopilotMetadata.RoutingInfo.(types.StakeibcPacketMetadata); ok &&
   routingInfo.Action == types.LiquidStake && routingInfo.IbcReceiver != "" {

    var err error
    tokenPacketData.Receiver, err = types.GenerateHashedAddress(packet.DestinationChannel, tokenPacketData.Sender)
    if err != nil {
        return channeltypes.NewErrorAcknowledgement(err)
    }
}
```

Reply:
Done [[here](https://github.com/Stride-Labs/stride/pull/1046/commits/4a8269211d5f90cfc34c3ecc109c5ff1aeb75473)]

Reply:
lgtm!